File: C++/Medium/WrongAnswer/maximum-subsequence-score_6.cpp
class Solution {
public:
    long long maxScore(vector<int>& nums1, vector<int>& nums2, int k) {
        
        const int n = nums1.size();
        vector<long long> sum1(k + 1, 0);
        vector<long long> sum2(k + 1, 0);
        for (int i = 0; i < k; ++i) {
            sum1[i + 1] = sum1[i] + nums1[i];
            sum2[i + 1] = sum2[i] + nums2[i];
        }
        long long ans = 0;
        for (int i = 0; i <= k; ++i) {
            int j = k - i;
            ans = max(ans, sum1[i] + sum2[j] - min(nums1[i], nums2[j]));
        }
        return ans;
    }
};

File: Java/Medium/RuntimeError/maximum-subsequence-score_6.java
class Solution {
    public long maxScore(int[] nums1, int[] nums2, int k) {
        
        long max = 0;
        int n = nums1.length;
        for (int i = 0; i <= k; i++) {
            int j = k - i;
            if (i > n || j > n) continue;
            long sum1 = 0;
            long sum2 = 0;
            for (int a = 0; a < i; a++) {
                sum1 += nums1[a];
            }
            for (int b = 0; b < j; b++) {
                sum2 += nums2[b];
            }
            max = Math.max(max, sum1 * Math.min(nums1[i - 1], nums2[j - 1]));
            for (int a = i, b = j; a < n && b < n; a++, b++) {
                sum1 += nums1[a];
                sum2 += nums2[b];
                max = Math.max(max, sum1 * Math.min(nums1[a], nums2[b]));
            }
        }
        return max;
    }
}